SU-HSDO/suhumsci
# Hsd8 80 course importer form

# READY FOR REVIEW

# Summary
- Built a form to allow a user to configure the course importer url.

# Needed By (Date)
- Thursday

# Urgency
- High

# Steps to Test

1. checkout this branch
1. sync archaeology `blt drupal:sync --site=archaeology --sync-files`
1. roll back courses `drush @archaeology.local mr hs_courses`
1. delete the key at `/admin/config/system/keys`
1. navigate to `/admin/structure/migrate/course-importer` page (its in the admin menu too)
1. provide at least 1 url from explorecourses.stanford.edu
1. navigate to `/admin/structure/migrate/manage/hs_courses/migrations` page and see the number of unprocessed items
1. add another url to the importer settings
1. validate the unprocessed items count increases.
1. run cron do use drush to import courses `drush @archaeology.local cron` or `drush @archaeology.local mim hs_courses`
1. validate correct courses were imported.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
